controller/resource_roles.go

```diff
@@ -87,6 +91,74 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
+
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if currentUser == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("identity ID not found in token"))
+	}
+
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+
+	hasScope, err := c.app.PermissionService().HasScope(ctx, *currentUser, ctx.ResourceID, authorization.ManageTeamsInSpaceScope)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	if !hasScope {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to assign roles"))
+	}
+
+	var identitiesToBeAssigned []uuid.UUID
+	for _, identity := range ctx.Payload.Data {
+		identityIDAsUUID, err := uuid.FromString(identity.ID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterError("identity", identity.ID).Expected("uuid"))
+		}
+		identitiesToBeAssigned = append(identitiesToBeAssigned, identityIDAsUUID)
+	}
+
+	// In batch assignment of roles all selected users must have previously been assigned
+	// privileges for the resource, otherwise the invitation workflow should be used instead
+	for _, identityIDAsUUID := range identitiesToBeAssigned {
+		assignedRoles, err := c.app.IdentityRoleRepository().FindIdentityRolesByIdentityAndResource(ctx, ctx.ResourceID, identityIDAsUUID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		if len(assignedRoles) == 0 {
+			log.Error(ctx, map[string]interface{}{
+				"resource_id": ctx.ResourceID,
+				"identity_id": identityIDAsUUID,
+				"role":        ctx.RoleName,
+			}, "identity not part of a  resource cannot be assigned a role")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewBadParameterErrorFromString("identityID", identityIDAsUUID, fmt.Sprintf("cannot update roles for an identity %s without an existing role", identityIDAsUUID)))
+		}
+	}
+
+	// Now that we have confirmed that all users have pre-existing role assignments
+	// we can proceed with the assignment of roles.
+	for _, identityIDAsUUID := range identitiesToBeAssigned {
+		err = c.app.RoleManagementService().Assign(ctx, identityIDAsUUID, ctx.ResourceID, ctx.RoleName)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"resource_id": ctx.ResourceID,
+				"identity_id": identityIDAsUUID,
+				"role":        ctx.RoleName,
+			}, "assignment failed")
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+	}
+
+	if err != nil {
```

Comment:
err can't be nil here, right?

Reply:
3a1ec42940e7a90357b68ba4c59c59cdfb0931d7 right, it gets checked above this already.